backend/controller/dal/encryption.go

```diff
@@ -84,5 +88,86 @@ func (d *DAL) EnsureKey(ctx context.Context, generateKey func() ([]byte, error))
 	}
 
 	logger.Debugf("Encryption key found, using it")
-	return encryptedKey, nil
+
+	return row.Key, nil
+}
+
+const verification = "FTL - Towards a 𝝺-calculus for large-scale systems"
+
+func (d *DAL) verifyEncryptor(ctx context.Context) (err error) {
+	var tx *Tx
+	tx, err = d.Begin(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to begin transaction: %w", err)
+	}
+	defer tx.CommitOrRollback(ctx, &err)
+
+	var row sql.GetOnlyEncryptionKeyRow
+	row, err = tx.db.GetOnlyEncryptionKey(ctx)
+	if err != nil {
+		if dal.IsNotFound(err) {
+			// No encryption key found, probably using noop.
+			return nil
+		}
+		return fmt.Errorf("failed to get encryption row from the db: %w", err)
+	}
+
+	needsUpdate := false
+	newTimeline, err := verifySubkey(d.encryptor, row.VerifyTimeline)
+	if err != nil {
+		return fmt.Errorf("failed to verify timeline subkey: %w", err)
+	}
+	if newTimeline != nil {
+		needsUpdate = true
+		row.VerifyTimeline = optional.Some(*newTimeline)
+	}
+
+	newAsync, err := verifySubkey(d.encryptor, row.VerifyAsync)
+	if err != nil {
+		return fmt.Errorf("failed to verify async subkey: %w", err)
+	}
+	if newAsync != nil {
+		needsUpdate = true
+		row.VerifyAsync = optional.Some(*newAsync)
+	}
+
+	if !needsUpdate {
+		return nil
+	}
+
+	if !row.VerifyTimeline.Ok() || !row.VerifyAsync.Ok() {
+		panic("should be unreachable. verifySubkey should have set the subkey")
+	}
+
+	err = tx.db.UpdateEncryptionVerification(ctx, row.VerifyTimeline, row.VerifyAsync)
+	if err != nil {
+		return fmt.Errorf("failed to update encryption verification: %w", err)
+	}
+
+	return nil
+}
+
+// verifySubkey checks if the subkey is set and if not, sets it to a verification string.
+// returns (nil, nil) if verified and not changed
+func verifySubkey[SK encryption.SubKey](encryptor encryption.DataEncryptor, encrypted optional.Option[encryption.EncryptedColumn[SK]]) (*encryption.EncryptedColumn[SK], error) {
```

Comment:
`encryption.EncryptedColumn` is a `[]byte` so `nil` will already work with it without having to make it a pointer.